### PR TITLE
SD-949 rename r_m_a to its fullname return_merchandise_authorization

### PR DESCRIPTION
**Pull request by KacperMekarski**
Reported here: https://github.com/spree/spree/issues/10456

The problem was that class RMARequired in classic reload was converted to rma_required to find a matching file, while zeitwerk required file named r_m_a_required. In my opinion best solution is to use fullname of rma (return_merchandise_authorization) which solves the problem of converting class to snake case in both cases.

Update: as brunovelasco mentioned, adding require_relative 'r_m_a_required' to core/app/models/spree/return_item/eligibility_validator/default.rb did not solve the problem.

I don't have any good idea how to test that using RSpec.

### CODE REVIEW

**Review by nnande — APPROVED** [review]

**Review by poudelprakash — APPROVED** [review]

**Review by damianlegawiec — COMMENTED** [review]
@KacperMekarski we need to fix this in a way we're not going to change the class name - this will break a lot of current spree instances, decorators and so on

**Review by piotrpanko — APPROVED** [review]

**Review by damianlegawiec — CHANGES_REQUESTED** [review]
@KacperMekarski 

This can be achieved by creating a `Zeitwerk::Inflector` initializer, eg. https://github.com/solidusio/solidus/blob/466ac8e4b3217dca86bc2194b3dcff56111c3e0c/core/config/initializers/inflections.rb#L3

**Comment by KacperMekarski** [comment]
@damianlegawiec done.

Works for both classic reload & zeitwerk mode

File: core/app/models/spree/return_item/eligibility_validator/default.rb [hunk 1]
```diff
@@ -1,5 +1,3 @@
-require_relative 'r_m_a_required'
```
Comment (KacperMekarski) [inline core/app/models/spree/return_item/eligibility_validator/default.rb:1]: I noticed that `require_relative` does the trick for file named `rma_required` (then adding acronym is not neccessary)

**Review by damianlegawiec — APPROVED** [review]

**Comment by KacperMekarski** [comment]
@damianlegawiec I rebased from `master`, but tests failed due to error from db migrations:
> Caused by:
NameError: uninitialized constant AddStockLocationToRMA

caused probably by added acronym.
Therefore I used second solution which I mentioned here: https://github.com/spree/spree/pull/10527#discussion_r520449256 which works fine.

**Comment by KacperMekarski** [comment]
Created new PR: https://github.com/spree/spree/pull/10574

**Pull request closed without merging**